### DISCUSSION TITLE
SSH keys for private repo access

### DISCUSSION THREAD

**Discussion by RusJJ** [community/community#58369] [Actions] [answered]
### Select Topic Area

Question

### Body

Cant get it working AT ALL. Lost one day of life getting "Host key verification failed.", LMAO. It just works PERFECTLY AMAZING on my PC.
In workflow:
>- shell: bash
>      env:
>        SSH_KEY: ${{ secrets.SSH }}
>      run: |
>        eval `ssh-agent -s`
>        echo "$SSH_KEY" > key
>        chmod 600 key
>        ssh-add key
>        git clone ssh://git@ssh.github.com:443/RusJJ/PrivateRepo.git

And yes, secret key SSH is correctly set. Tried it few times. Key is PEM, starts with -----BEGIN RSA PRIVATE KEY----- (telling this to stop time wasting questions, sorry!)
Please, help me...

**Comment by airtower-luna** [comment]
> And yes, secret key SSH is correctly set. Tried it few times. Key is PEM, starts with -----BEGIN RSA PRIVATE KEY----- (telling this to stop time wasting questions, sorry!)

That's probably the issue, actually. OpenSSH uses its own file format (which is kinda PEM-like), and the first line should be:

```
-----BEGIN OPENSSH PRIVATE KEY-----
```

And then the format is different, just changing that line won't work, you either need to export your key in OpenSSH format and store that in the secret, or use `ssh-keygen` in your workflow to convert the key you have to the OpenSSH format. :wink: 

As for the SSH host keys, you can retrieve them from the "meta" API endpoint (see the `ssh_keys` attribute): https://api.github.com/meta

**Reply by RusJJ** [reply]
Encrypted PEM key has worked on my own PC. Thus OPENSSH key gives me exactly the same error:
![image](https://github.com/community/community/assets/8864329/8ae1d892-dad7-4f60-b8f5-a68a8b28535c)

**Reply by airtower-luna** [reply, marked as answer]
It's possible you have an old OpenSSH version or a different client locally.

For the host key problem you need to use the API I mentioned above, and add the returned keys (not the whole JSON, just the keys, one per line) to the `~/.ssh/known_hosts` file.

**Reply by RusJJ** [reply]
Alright, i tried to use webfactory/ssh-agent
It gives me this thingo: Comment for (public) key 'ssh-rsa balblabla== (stdin)' does not match GitHub URL pattern. Not treating it as a GitHub deploy key.

How ..?

**Reply by RusJJ** [reply]
Also tried the thing with known_hosts file at it didnt work at all...

**Reply by WFig7** [reply]
How to hack a Snapchat account Give me a link to hack the account 📴 now

**Reply by RusJJ** [reply]
> How to hack a Snapchat account Give me a link to hack the account 📴 now

Ask your family

**Comment by RusJJ** [comment]
Alright, i used it and... i found something interesting. That was the problem, and i realised it only now. I was trying to get an access to my repo through the **ssh://git@ssh.github.com:443**
DONT do that. Just do **ssh://git@github.com**
It works on my PC but doesnt in Actions. Seems like it works now!

**Reply by gahlawatriya827-bit** [reply]
How I can hack another insta account

**Comment by hammojokr-eng** [comment]
اريد الهجوم علي موسسه عسركيه الي الجيش الإسرائيلي
